UI/EnemyTree.gd

```diff
@@ -30,16 +30,35 @@ func _on_FilterLineEdit_text_changed(new_text):
 	_rebuild_list(new_text)
 	
 func _rebuild_list(filter_text: String = ""):
-	var normalized_filter_text := filter_text.to_upper()
+	var normalized_filter_text = filter_text.to_upper()
+	var filter_int = int(filter_text)
```

Comment:
if aiming for performance, convert to int outside the for loop, but also store in variables the results of filter_text.is_valid_integer() and filter_text.is_valid_hex_number(true) so they get run only once
if aiming for readability, do the conversion inside the elif or at least inside the for